### ISSUE TITLE
splash:runjs returns a table, not a string in case of failure

### ISSUE THREAD

**Issue by lopuhin** [scrapinghub/splash#610]
According to the [docs](http://splash.readthedocs.io/en/stable/scripting-ref.html#splash-runjs), ``splash:runjs`` should return a string as the second return value in case of error. But for me it returns a table. Here is an example script:
```
function main(splash)
  local url = "http://google.com"
  assert(splash:go(url))
  assert(splash:wait(0.5))
  ok, reason = splash:runjs("document.getElementById('foo').click(); ")
  
  json = require('json')
  assert(ok, json.encode(reason))
  return {
    html = splash:html(),
    png = splash:png(),
    har = splash:har(),
  }
end
```

And here is the output:
```
Lua error: [string "function main(splash)
..."]:8: {"js_error_type": "TypeError", "type": "JS_ERROR", "splash_method": "runjs", "js_error_message": "null is not an object (evaluating 'document.getElementById('foo').click')", "js_error": "TypeError: null is not an object (evaluating 'document.getElementById('foo').click')", "message": "JS error: \"TypeError: null is not an object (evaluating 'document.getElementById('foo').click')\""}
```

Due to this, I can't use the usual ``assert(splash:runjs)`` pattern. If I try to just pass ``reason`` to assert, I get:

```
Lua error: [string "function main(splash)
..."]:8: bad argument #2 to 'assert' (string expected, got table)
```

**Comment by kmike** [comment]
The behavior was changed in https://github.com/scrapinghub/splash/pull/307, but for no apparent reason, and table doesn't play well with assert indeed.

**Comment by kmike** [comment]
For what it worths, Lua 5.3 should fix this issue - `assert` can work if a second argument is a table. But Lua 5.3 support is not implemented in lupa. `error` can raise any Lua object as well, not only a string, even in Lua 5.2. 

So I guess we should fix it for now, and then investigate reverting the fix and switching to Lua 5.3.

**Comment by lopuhin** [comment]
Thanks @kmike !
